### PR TITLE
[OneCollectorExporter] Add json formatter support

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Adds support for custom complex type serialization into json
* Extends the transmission callback mechanism so users can op-into receiving failed transmissions

## TODOs

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OneCollector/CommonSchema/CommonSchemaMetadataFieldPrivacyClassificationType.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.OneCollector;
+
+/// <summary>
+/// Describes the Common Schema field privacy classification type.
+/// </summary>
+public enum CommonSchemaMetadataFieldPrivacyClassificationType
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OneCollector/CommonSchema/CommonSchemaMetadataFieldPrivacyClassificationType.cs:22]: Why do we want these in the exporter?

Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.OneCollector/CommonSchema/CommonSchemaMetadataFieldPrivacyClassificationType.cs]: They are necessary if we want to make a really friendly API. I just pushed a change which drops them and instead exposes the underlying computed value. That will push the heavy lifting for sorting out how to compute the value onto consumers. Works for me if you would prefer it that way.

Comment (reyang) [reply src/OpenTelemetry.Exporter.OneCollector/CommonSchema/CommonSchemaMetadataFieldPrivacyClassificationType.cs]: I think privacy classification is independent of exporter and should be handled elsewhere. More importantly, OpenTelemetry's philosophy is that instrumentation should be agnostic to backend.

File: src/OpenTelemetry.Exporter.OneCollector/.publicApi/net7.0/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporter<T>.RegisterPayloadTransmittedCallback(OpenTelemetry.Exporter.OneCollector.OneCollectorExporterPayloadTransmittedCallbackAction! callback, bool includeFailures) -> System.IDisposable?
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterJsonSerializationOptions
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterJsonSerializationOptions.OneCollectorExporterJsonSerializationOptions() -> void
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterJsonSerializationOptions.RegisterFormatter<T>(System.Action<T, System.Text.Json.Utf8JsonWriter!>! formatter) -> OpenTelemetry.Exporter.OneCollector.OneCollectorExporterJsonSerializationOptions!
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterPayloadTransmittedCallbackArguments.Succeeded.get -> bool
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterSerializationOptions
+OpenTelemetry.Exporter.OneCollector.OneCollectorExporterSerializationOptions.JsonOptions.get -> OpenTelemetry.Exporter.OneCollector.OneCollectorExporterJsonSerializationOptions!
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OneCollector/.publicApi/net7.0/PublicAPI.Unshipped.txt:7]: Do we want to expose anything that's tight to the actual wire protocol (JSON)?

**Pull request closed without merging**